### PR TITLE
Fix errors to pass validator [GSTV-206]

### PR DESCRIPTION
**Pull request by Vergil333**
https://roihunter.atlassian.net/browse/GSTV-206

### CODE REVIEW

**Comment by Vergil333** [comment]
@roma-kisel @martinfryc1 

1. rest of the errors could be solved by removing filex:

- views/templates/admin/products.php
- classes/ZboziAttributes.php

I think those files are not used anywhere. That will solve almost all errors.

2. Then the constants in case statements should be replaced by plain string (because validator is too dumb to get them).

3. "The use of function base64_encode() is forbidden" we have to replace this function.

**Comment by Vergil333** [comment]
I removed those files and tested plugin. Works good, it was not needed.

**Comment by Vergil333** [comment]
All the errors and warnings trowed by a Validator has been fixed.
[roihunter.zip](https://github.com/roihunter/roi-hunter-easy-presta-plugin/files/3290441/roihunter.zip)
errors_and_warning_are_fixed

File: views/js/back.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-*  @author    PrestaShop SA <contact@prestashop.com>
-*  @copyright 2007-2018 PrestaShop SA
-*  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
-*  International Registered Trademark & Property of PrestaShop SA
-*
-* Don't forget to prefix your containers with your own identifier
-* to avoid any conflicts with others containers.
-*/
+/**
+* 2007-20201918 PrestaShop
```
Comment (roma-kisel) [inline views/js/back.js]: 20201918 :thinking:

File: classes/ZboziAttributes.php [hunk 2]
```diff
@@ -11,8 +11,8 @@
  *
  * @package    zbozi
  * @author    Vaclav Mach <info@prestahost.cz>
- * @copyright 2014,2015 Vaclav Mach
  * @license   EULA
+ * @copyright 2014,2019 Vaclav Mach
```
Comment (martinfryc1) [inline classes/ZboziAttributes.php]: It looks like replace mistake.

**Comment by martinfryc1** [comment]
@oparizek Could you please add your final review? Thanks

File: state.php [hunk 3]
```diff
@@ -28,9 +39,7 @@
 
     header("HTTP/1.1 200 OK");
     die();
-
-} else if ($_SERVER['REQUEST_METHOD'] == 'GET') {
-
+} elseif ($_SERVER['REQUEST_METHOD'] == 'GET') {
```
Comment (roma-kisel) [inline state.php]: else <should be space> if

File: classes/ProductJson.php [hunk 4]
```diff
@@ -31,118 +33,58 @@ public function getJson($id_product, $id_product_attribute, $id_lang, $id_shop)
         return $this->getJsonRow($product, $id_product_attribute);
     }
 
+    private function getSingleProduct($id_product, $id_lang, $id_shop)
```
[the remark this answers is not in the document]
Comment (Vergil333) [reply classes/ProductJson.php]: Probably not. There were hundreds of errors which I solved by [php-cs-fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/tree/master) and this script moved the method, but solved a lot of the errors.

File: state.php (hunk 3 above)
Comment (Vergil333) [reply state.php]: By the PSR2 standard both should be elseif. Together.

File: classes/ProductJson.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (Vergil333) [reply classes/ProductJson.php]: Yup, the php-cs-fixer is also ordering methods by the function **ordered_class_elements**.

If you look at function **getJson** in **ProductJson.php** file, you can see what methods it calls and those methods are later in exact order to this function's calls.

**Pull request merged**